resources/migrations/000_migrations.yaml

```diff
@@ -7653,3 +7653,85 @@ databaseChangeLog:
             referencedColumnNames: id
             constraintName: fk_view_log_ref_user_id
             onDelete: CASCADE
+
+# changesets 268-272 allow for handling user account emails in lowercase (GH issue 3047)
+  - property:
+      name: email_constraint_name
+      value: constraint_4
+      dbms: h2
+  - property:
+      name: email_constraint_name
+      value: core_user_email_key
+      dbms: postgresql
+  - property:
+      name: email_constraint_name
+      value: email
+      dbms: mysql,mariadb
+  - changeSet:
+      id: 268
+      author: rlotun
+      comment: Added 0.37.0   # drop unique constraint on email - this will also remove supporting indexes
+      preConditions:
+          - or:
+              - dbms:
+                    type: mysql
+              - dbms:
+                    type: mariadb
+              - dbms:
+                    type: postgresql
+              - dbms:
+                    type: h2
+      changes:
+        - dropUniqueConstraint:
+            columnDataType: varchar(254)
+            constraintName: ${email_constraint_name}
+            tableName: core_user
+  - changeSet:
+      id: 269
+      author: rlotun
+      comment: Added 0.37.0  # set email to lower email
+      changes:
+        - sql :
+            sql : update core_user set email = lower(email)
+  - property:
+      name: email_c_type
+      value: varchar_ignorecase(254)
+      dbms: h2
+  - changeSet:
+      id: 270
+      author: rlotun
+      comment: Added 0.37.0 # for H2 convert column to VARCHAR IGNORECASE=TRUE (PG needs citext extension installed)
+      failOnError: false
+      preConditions:
+          - or:
+              - dbms:
+                    type: h2
+      changes:
+        - modifyDataType:
+            tableName: core_user
+            columnName: email
+            newDataType: ${email_c_type}
+  - changeSet:
+      id: 271
+      author: rlotun
+      comment: Added 0.37.0  # add unique constraint on lower(email)
+      changes:
+        - addUniqueConstraint:
+            columnNames: email
```

Comment:
it looks like this constraint is on `email`, not `lower(email)`

Reply:
as discussed in slack, you can't have function based constraints. we're going to leave this constraint on email and add the lower(email) index as well (it seems modern dbs will create a backing indexes on unique constraints anyway).